server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -847,11 +847,35 @@ public void forceMerge(boolean flush) throws IOException {
      */
     public abstract IndexCommitRef acquireSafeIndexCommit() throws EngineException;
 
+    /**
+     * If the specified throwable contains a fatal error in the throwable graph, such a fatal error will be thrown. Callers should ensure
+     * that there are no catch statements that would catch an error in the stack as the fatal error here should go uncaught and be handled
+     * by the uncaught exception handler that we install during bootstrap. If the specified throwable dies indeed contain a fatal error, the
```

Comment:
I think you mean "does" (indeed contain a fatal error) on this line.

Reply:
Thank you @richardun, I appreciate the attention to detail. I will push a correction in the morning.